fenix/app/src/main/java/org/mozilla/fenix/tabhistory/TabHistoryDialogFragment.kt

```diff
@@ -1,72 +0,0 @@
-/* This Source Code Form is subject to the terms of the Mozilla Public
- * License, v. 2.0. If a copy of the MPL was not distributed with this
- * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
-
-package org.mozilla.fenix.tabhistory
-
-import android.os.Bundle
-import android.view.LayoutInflater
-import android.view.View
-import android.view.ViewGroup
-import androidx.navigation.fragment.findNavController
-import com.google.android.material.bottomsheet.BottomSheetBehavior
-import com.google.android.material.bottomsheet.BottomSheetDialog
-import com.google.android.material.bottomsheet.BottomSheetDialogFragment
-import kotlinx.coroutines.flow.distinctUntilChanged
-import kotlinx.coroutines.flow.mapNotNull
-import mozilla.components.browser.state.selector.findCustomTabOrSelectedTab
-import mozilla.components.lib.state.ext.flowScoped
-import mozilla.components.support.ktx.android.content.getColorFromAttr
-import org.mozilla.fenix.R
-import org.mozilla.fenix.databinding.FragmentTabHistoryDialogBinding
-import org.mozilla.fenix.ext.requireComponents
-
-class TabHistoryDialogFragment : BottomSheetDialogFragment() {
```

Comment:
Sorry for taking so long to get to this review. I have some querying questions to better understand how you're approaching this. I was wondering if we could still keep the fragment navigation and implement this bottom sheet similar to:
https://searchfox.org/mozilla-mobile/source/firefox-android/fenix/app/src/main/java/org/mozilla/fenix/shopping/ReviewQualityCheckFragment.kt
https://searchfox.org/mozilla-mobile/source/firefox-android/fenix/app/src/main/java/org/mozilla/fenix/translations/TranslationsDialogFragment.kt

Reply:
I made this refactoring because it was a crash at navigation and we don't have steps to reproduce it. https://bugzilla.mozilla.org/show_bug.cgi?id=1804377
If I add navigation to this refactoring, it is almost useless because the crash still happens.